universite-du-quebec-a-montreal.csl

```diff
@@ -1,487 +1,542 @@
-              <names variable="editor">
-                <label form="verb-short" text-case="capitalize-first" suffix=". "/>
-                <name and="text" delimiter=", "/>
-              </names>
-            </if>
-          </choose>
-          <choose>
-            <if variable="author editor" match="any">
-              <names variable="translator">
-                <label form="verb-short" text-case="capitalize-first" suffix=" "/>
-                <name and="text" delimiter=", "/>
-              </names>
-            </if>
-          </choose>
-        </group>
-      </if>
-    </choose>
-  </macro>
   <macro name="container-contributors">
     <choose>
-      <if type="chapter paper-conference" match="any">
-        <group prefix=", " delimiter=", ">
-          <choose>
-            <if variable="author">
-              <names variable="editor">
-                <label form="verb-short" prefix=" " text-case="lowercase" suffix=" "/>
-                <name and="text" delimiter=", "/>
-              </names>
-              <choose>
-                <if variable="container-author">
-                  <group>
-                    <names variable="container-author">
-                      <label form="verb-short" prefix=" " text-case="lowercase" suffix=" "/>
-                      <name and="text" delimiter=", "/>
-                    </names>
-                  </group>
-                </if>
-              </choose>
-            </if>
-          </choose>
-          <choose>
-            <if variable="author editor" match="any">
-              <names variable="translator">
-                <label form="verb-short" prefix=" " text-case="lowercase" suffix=" "/>
-                <name and="text" delimiter=", "/>
-              </names>
-            </if>
-          </choose>
-        </group>
+      <if type="chapter paper-conference entry-encyclopedia entry-dictionary" match="any">
+        <names variable="editor container-author" delimiter=", " suffix=", ">
+          <name and="text" initialize-with=". " delimiter=", "/>
+          <label form="short" prefix=" (" text-case="title" suffix=")"/>
+          <label form="short" prefix=" (" suffix=")" text-case="lowercase"/>
+        </names>
       </if>
     </choose>
   </macro>
-  <macro name="anon">
+  <macro name="secondary-contributors">
     <choose>
-      <if variable="author editor translator" match="none">
-        <text term="anonymous" form="short" text-case="capitalize-first" suffix="." strip-periods="false"/>
+      <if type="article-journal book chapter paper-conference entry-encyclopedia" match="none">
+        <names variable="editor" delimiter=", " prefix=" (" suffix=")">
+          <name name-as-sort-order="all" and="text" sort-separator=", " initialize-with=". " delimiter=", " delimiter-precedes-last="never"/>
+          <label form="short" prefix=", " text-case="title"/>
+          <label form="short" prefix=" (" suffix=")" text-case="lowercase"/>
```

Comment:
sorry, I didn't see the ping. The lowercase labels in parentheses are correct both here and above